interpreter/runtime/memory.mli

```diff
@@ -43,3 +43,6 @@ val load_simd_lane :
 val store_packed :
   pack_size -> memory -> address -> offset -> value -> unit
     (* raises Type, Bounds *)
+val store_simd_lane :
```

Comment:
I wonder if it wouldn't make more sense to remove these lane functions from the Memory module and refactor slightly: turn the splicing part into functions in the Simd module and perform the memory access separately in Eval via the existing load/store_packed functions.

Reply:
store_simd_lane is: extract_lane + store_value/store_packed.
extract_lane (and correspondingly replace_lane for load_simd_lane) is already defined in Simd module.
We could remove these lane functions from Memory, it would probably look like inlining what is already in store_simd_lane, since the bulk of the work is deciding which extract_lane functions to call for various pack_size.
Or instead of inlining, we define a small helper in Simd, to perform this mapping from pack_size -> extract_lane functions.

Reply:
Yes, the latter is more or less what I had in mind. That could go into the Simd module.

Reply:
Happy a bit of difficulty defining the helper.
The extract lane functions have different signatures that aren't compatible:
`i32x4.extract_lane` returns a int32, `i64x2.extract_lane` returns a int64.
And I can't wrap those in a `value` due to circular dependency (`Values` depends on `Simd` already).
The best I can do right now is to inline the logic inside of eval.ml.

Reply:
Then I'd inline it in eval.ml. It's only used once anyway, isn't?

Reply:
Yup, only used once, I inlined them into SimdStoreLane, ptal.

Reply:
For consistency, can you do the same for simd_load_lane?

Reply:
Done!